### PR TITLE
SWC-6207: Update Dataset icon, and add DatasetCollection

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/lib/utils/SynapseClient.ts [hunk 1]
```diff
@@ -1374,10 +1374,10 @@ export const setAccessTokenCookie = async (
       // See - https://github.com/reactivestack/cookies/issues/189
       await delay(100)
     } else {
-      // set's cookie in session storage
+      // sets cookie
```
Comment (jay-hodgson) [inline src/lib/utils/SynapseClient.ts:1377]: For local testing only (not accessed in a production environment).

**Comment by jay-hodgson** [comment]
New Dataset and Dataset Collection icons (and add support for DS collection in EntityIcon):
Screen Shot 2022-07-12 at 11 03 45 AM

File: src/lib/containers/IconSvg.tsx [hunk 2]
```diff
@@ -374,7 +377,9 @@ const getIcon = (options: IconSvgOptions) => {
     case 'discussion':
       return <QuestionAnswerTwoTone style={muiSvgStyle} />
     case 'dataset':
-      return <GridOnTwoTone style={muiSvgStyle} />
```
Comment (yjyoo773) [inline src/lib/containers/IconSvg.tsx:377]: Is the GridOnTwoTone being used elsewhere or can we remove it from import?

Comment (jay-hodgson) [reply src/lib/containers/IconSvg.tsx:377]: Good call, probably not used.  I'll check

**Review by yjyoo773 — APPROVED** [review]

**Pull request merged**